Explorer/Assets/Scripts/MVC/MVCManager.cs

```diff
@@ -0,0 +1,170 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using MVC.PopupsController.PopupCloser;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.AddressableAssets;
+
+namespace MVC
+{
+    public class MVCManager : IMVCManager
+    {
+        internal readonly Dictionary<Type, IController> controllers;

```

Comment:
Let's expose the dictionary as `IReadOnlyDictionary`. This way it looks dirty

Reply:
we call add in the RegisterController function, we cannot have this as readonly

Reply:
I mean you keep Dictionary but as a private field and **expose** it publicly as `IReadOnlyDictionary`